goawk.go

```diff
@@ -245,6 +278,20 @@ argsLoop:
 		errorExitf("%s", err)
 	}
 
+	coverageHelper := cover.New(covermode, coverappend, fileReader)
+	if covermode != "" {
+		astProgram := &prog.ResolvedProgram.Program
+		coverageHelper.Annotate(astProgram)
+		prog, err = parser.ResolveAndCompile(astProgram, parserConfig)
+		if err != nil {
+			errorExitf("%s", err)
+		}
+		if coverprofile == "" {
```

Comment:
Well, the one thing I really dislike about `-d` is that not only it outputs the syntax tree, but **also runs the code**.
To me, you should alsways do one thing at a time. That is, when I want to check that the code is parsed and/or annotated for coverage correctly **I do not want it to also run**. 
So I propose:
1. Either we change the behavior of `-d` to not run the code and drop the "unintuitive behavior"
2. Or we keep this as is due to the above reason above.

Reply:
Very fair, I've often thought that it shouldn't run the program after the `-d*` options. How about you put up another PR to change that, then merge it into this. I think it's okay to change the behaviour of those as they're just debugging commands. Let's do this:

* Make `-d`, `-dt`, and `-da` print to stdout instead of stderr.
* If any of the above are specified, make GoAWK exit immediately afterwards.
* Separate out the help output for the debugging command into another group, like so:

```go
	longUsage  = `Standard AWK arguments:
  -F separator      field separator (default " ")
  -f progfile       load AWK source from progfile (multiple allowed)
  -v var=value      variable assignment (multiple allowed)

Additional GoAWK features:
  -E progfile       load program, treat as last option, disable var=value args
  -H                parse header row and enable @"field" in CSV input mode
  -h, --help        show this help message
  -i mode           parse input into fields using CSV format (ignore FS and RS)
                    'csv|tsv [separator=<char>] [comment=<char>] [header]'
  -o mode           use CSV output for print with args (ignore OFS and ORS)
                    'csv|tsv [separator=<char>]'
  -version          show GoAWK version and exit

GoAWK debugging arguments:
  -cpuprofile file  write CPU profile to file
  -d                print parsed syntax tree to stdout and exit
  -da               print VM assembly instructions to stdout and exit
  -dt               print variable type information to stdout and exit
  -memprofile file  write memory profile to file
`
```

Note I'm also adding in help for the `-memprofile` option, which I guess I'd missed before. And then of course the new `-cover*` options would go into the debugging section.

Reply:
Do you think we need to add some tests for the `-d*` options?

Reply:
Excellent -- approved and merged.